runtime/j9vm/java11vmi.c

```diff
@@ -710,18 +730,80 @@ allowReadAccessToModule(J9VMThread * currentThread, J9Module * fromModule, J9Mod
  * @return If successful, returns a java.lang.reflect.Module object. Otherwise, returns NULL.
  */
 jobject JNICALL
+#if JAVA_SPEC_VERSION >= 15
+JVM_DefineModule(JNIEnv * env, jobject module, jboolean isOpen, jstring version, jstring location, jobjectArray packageArray)
+#else
 JVM_DefineModule(JNIEnv * env, jobject module, jboolean isOpen, jstring version, jstring location, const char* const* packages, jsize numPackages)
+#endif /* JAVA_SPEC_VERSION >= 15 */
 {
 	J9VMThread * const currentThread = (J9VMThread*)env;
 	J9JavaVM * vm = currentThread->javaVM;
 	J9InternalVMFunctions const * const vmFuncs = vm->internalVMFunctions;
+#if JAVA_SPEC_VERSION >= 15
+	BOOLEAN oom = FALSE;
+	BOOLEAN nullPackage = FALSE;
+	jsize numPackages = (NULL == packageArray) ? 0 : (*env)->GetArrayLength(env, packageArray);
+	PORT_ACCESS_FROM_ENV(env);
+	UDATA packagesNumBytes = sizeof(char*) * numPackages;
+	const char* * packages = (const char* *)j9mem_allocate_memory(packagesNumBytes, OMRMEM_CATEGORY_VM);
+	if ((*env)->ExceptionCheck(env)) {
+		goto cleanup;
+	}
+
+	if (NULL != packages) {
+		jsize pkgIndex = 0;
+		memset(packages, 0, packagesNumBytes);
```

Comment:
This fails to compile on Windows:
```
12:44:56  java11vmi.c(756): warning C4090: 'function': different 'const' qualifiers
```
This will need a cast:
```
    memset((void *)packages, 0, packagesNumBytes);
```